### PR TITLE
Implement Attestation Participants Helper Function

### PR DESCRIPTION
**Pull request by terenc3t**
This PR implements attestation participants helper function as outlined [here](https://github.com/ethereum/eth2.0-specs/blob/dbf46f6f4775f9fa41b636a62d1a7d36e0ec4a02/specs/core/0_beacon-chain.md#get_attestation_participants)

This helper function is needed to move further on epoch processing

### CODE REVIEW

File: beacon-chain/core/validators/committees.go [hunk 1]
```diff
@@ -102,3 +103,64 @@ func getCommitteesPerSlot(numActiveValidatorRegistry uint64) uint64 {
...
+
+	var participants *pb.ShardAndCommittee
+	for _, committee := range shardCommittees.ArrayShardAndCommittee {
+		if committee.Shard == attestationData.Shard {
+			participants = committee
+			break
+		}
+	}
+
+	if len(participationBitfield) != len(participants.Committee) {
```
Comment (prestonvanloon) [inline beacon-chain/core/validators/committees.go]: Spec says: `assert len(participation_bitfield) == ceil_div8(len(shard_committee.committee))`

Why is this different?

Comment (terenc3t) [reply beacon-chain/core/validators/committees.go]: Changed it to use `math.Ceil` for `ceil_div8(len(shard_committee.committee))`

File: beacon-chain/core/validators/committees.go [hunk 2]
```diff
@@ -102,3 +104,64 @@ func getCommitteesPerSlot(numActiveValidatorRegistry uint64) uint64 {
...
+
+	var participants *pb.ShardAndCommittee
+	for _, committee := range shardCommittees.ArrayShardAndCommittee {
+		if committee.Shard == attestationData.Shard {
+			participants = committee
+			break
+		}
+	}
+
+	if float64(len(participationBitfield)) != math.Ceil(float64(len(participants.Committee))) {
```
Comment (rauljordan) [inline beacon-chain/core/validators/committees.go]: im worried using floats will eventually lead to nasty errors - @rawfalafel expressed similar concerns in some past PRs especially if we're doing them for comparison. We should find an alternative to this.

File: beacon-chain/core/validators/committees.go [hunk 3]
```diff
@@ -102,3 +104,64 @@ func getCommitteesPerSlot(numActiveValidatorRegistry uint64) uint64 {
...
+			"wanted participants bitfield length %d, got: %d",
+			len(participants.Committee), len(participationBitfield))
+	}
+
+	// Find the participating attesters in the committee.
+	var participantIndices []uint32
+	for i, validatorIndex := range participants.Committee {
+		bitSet, err := bitutil.CheckBit(participationBitfield, i)
+		if err != nil {
+			return nil, err
```
Comment (rauljordan) [inline beacon-chain/core/validators/committees.go]: Can you do an errorf to give some more color to this failing?

File: beacon-chain/core/validators/committees.go [hunk 4]
```diff
@@ -102,3 +104,64 @@ func getCommitteesPerSlot(numActiveValidatorRegistry uint64) uint64 {
...
+
+	var participants *pb.ShardAndCommittee
+	for _, committee := range shardCommittees.ArrayShardAndCommittee {
+		if committee.Shard == attestationData.Shard {
+			participants = committee
+			break
+		}
+	}
+
+	if len(participationBitfield) != int(math.Ceil(float64(len(participants.Committee)))) {
```
Comment (prestonvanloon) [inline beacon-chain/core/validators/committees.go]: I still think this is wrong. 

```python
assert len(participation_bitfield) == ceil_div8(len(shard_committee.committee))
```

Doesn't `ceil_div8` mean: "Divide by 8 and take the ceiling of that number"? Casting an int the a float and taking the ceiling of that float then casting back to int effectively does nothing.

Comment (prestonvanloon) [reply beacon-chain/core/validators/committees.go]: https://kite.com/python/docs/Crypto.Util.number.ceil_div

Comment (terenc3t) [reply beacon-chain/core/validators/committees.go]: Ah. That's correct. I have no idea what I was thinking...

Updated it to: `int(math.Ceil(float64(len(participants.Committee)/8)))`

File: shared/mathutil/math_helper_test.go [hunk 5]
```diff
@@ -40,3 +38,38 @@ func TestIntegerSquareRoot(t *testing.T) {
...
+		{
+			number: 97282,
+			div8:   12161,
+		},
+	}
+
+	for _, tt := range tests {
+		div8 := CeilDiv8(tt.number)
+		if tt.div8 != div8 {
+			t.Fatalf("Div8 was an expected value. Wanted: %d, got: %d", tt.div8, div8)
```
Comment (prestonvanloon) [inline shared/mathutil/math_helper_test.go]: Typo: should be “was not”

File: beacon-chain/core/validators/committees.go [hunk 6]
```diff
@@ -102,3 +103,64 @@ func getCommitteesPerSlot(numActiveValidatorRegistry uint64) uint64 {
...
+	}
+
+	if len(participationBitfield) != mathutil.CeilDiv8(len(participants.Committee)) {
+		return nil, fmt.Errorf(
+			"wanted participants bitfield length %d, got: %d",
+			len(participants.Committee), len(participationBitfield))
+	}
+
+	// Find the participating attesters in the committee.
+	var participantIndices []uint32
```
Comment (nisdas) [inline beacon-chain/core/validators/committees.go]: Maybe make this 
```go
participantIndices := make([]uint32,0,len(participants.Committee))
```

**Pull request merged**